src/vm/jitinterface.cpp

```diff
@@ -2298,7 +2300,8 @@ unsigned CEEInfo::getClassGClayout (CORINFO_CLASS_HANDLE clsHnd, BYTE* gcPtrs)
             {
                 // Get offset into the value class of the first pointer field (includes a +Object)
                 size_t cbSeriesSize = pByValueSeries->GetSeriesSize() + pMT->GetBaseSize();
-                size_t cbOffset = pByValueSeries->GetSeriesOffset() - OBJECT_SIZE;
+                size_t cbSeriesOffset = pByValueSeries->GetSeriesOffset();
```

Comment:
This layout is not version resilient. This should have assert and/or throw platform not supported that will make sure that the JIT won't ever get the non-resilient layout when compiling for R2R.

Reply:
I added an assert here and also in getHeapClassSize.